### PR TITLE
nixos/fcitx5: add settings

### PR DESCRIPTION
**Pull request by linsui**
###### Description of changes

Add global settings for fcitx5. User config files overrides settings in `/etc/xdg/fcitx5`.

Addon settings are ini files with global section. The type will be added in https://github.com/NixOS/nixpkgs/pull/228051.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/i18n/input-method/fcitx5.nix [hunk 1]
```diff
@@ -30,11 +69,30 @@ in
   config = mkIf (im.enabled == "fcitx5") {
     i18n.inputMethod.package = fcitx5Package;
 
+    environment.etc =
+      let
+        optionalFile = p: f: v: lib.optionalAttrs (v != { }) {
+          "xdg/fcitx5/${p}".text = f v;
```
Comment (Vonfry) [inline nixos/modules/i18n/input-method/fcitx5.nix:75]: Can the system wide XDG_CONFIG_HOME be changed by environment variable or other method? Is there a better way to set it? Or can you read fcitx5 to check how it determine this file path such as reading from hard coded path or from others.

Comment (linsui) [reply nixos/modules/i18n/input-method/fcitx5.nix:75]: i'm not sure what you mean. Do you want to ask why this works? The related code is https://github.com/fcitx/fcitx5/blob/master/src/lib/fcitx-utils/standardpath.cpp#L81.

Comment (Vonfry) [reply nixos/modules/i18n/input-method/fcitx5.nix:75]: Allright, `/etc/xdg` is hard-coded in the source instead of some variables.

**Review by Vonfry — APPROVED** [review]

**Comment by NickCao** [comment]
Please resolve the conflicts.

**Comment by NickCao** [comment]
This breaks evaluation. `undefined variable 'sts'`

**Comment by NickCao** [comment]
Fixed in https://github.com/NixOS/nixpkgs/pull/249223

**Comment by linsui** [comment]
Sorry, a line is missed in the rebase. Thanks for your fix!

**Pull request merged**